lib/gears/shape.back

```diff
@@ -0,0 +1,449 @@
+---------------------------------------------------------------------------
```

Comment:
Uhm, I think something went wrong with this commit ("add radial_progress shape"). It adds way more stuff than what the commit messages makes me expect.

Should this PR be split into two parts, one addings lots of new shapes (basically the first few commits) and another one handles the new "magic doc-images"?

(GitHub's interface changed, I can't comment at the bottom of this commit... Let's see where this comment ends up)